current/standard-derivatives.md

```diff
@@ -56,7 +56,34 @@ sys.exit
 
 ### Scala DF
 
-TODO
+```scala
+import io.archivesunleashed._
+import io.archivesunleashed.df._
+
+sc.setLogLevel("INFO")
+
+// Web archive collection.
+val warcs = RecordLoader.loadArchives("/path/to/data", sc)
+  .webpages()
+
+// Domains file.
+warcs.groupBy(ExtractDomainDF($"Url").alias("url"))
```

Comment:
Shouldn't there be a `select` somewhere here so you throw away all the other fields from `webpages`? Would it make sense to print out the schema of the derivatives?

Reply:
> Don't need to do a `select` if you're doing `groupBy` and `count` to get the `count` column. If you do just a `select` and `count`, you only count the rows iirc.

Neat, didn't know that!

> As for schema, we're replicating what we're doing in the RDD version of the script we run in `auk`, which is directly above this, and we don't do that there. If you're talking about including the CSV headers, that's another thing. Though, when you `cat` all the part files together, you have CSV headers in each one, and you'd have to filter them all out.

Okay, sg.